### PR TITLE
feat(cmd/celestia): Add `auth` subcommand to bridge, full and light

### LINKED ISSUE #1187: add authorization
- [ ] scaffolding for perms
- [ ] jwt token gen (in node module)
- [ ] add header and token in client wrapper 
- [ ] add auth methods on node module

### PR DESCRIPTION
**Pull request by renaynay**
Resolves #1187

This PR adds an `auth` subcommand to `bridge`, `full` and `light` nodes. 

Intended use: after a node is already initialized and started, a user can use the CLI to generate a signed token with some given perms via the `auth` cmd.

Example:

```
celestia bridge auth admin 
```

The above command outputs a signed JWT token with admin permissions that a user can use with `client.NewClient()` to access admin-level methods on the node.

Reference [this commit](https://github.com/celestiaorg/celestia-node/pull/1429/commits/2c833bca89790ff5503822875751f3330207b87f) for the actual diff.

### CODE REVIEW

**Review by distractedm1nd — DISMISSED** [review]

File: cmd/celestia/bridge.go [hunk 1]
```diff
@@ -35,6 +35,15 @@ func init() {
...
 		),
+		authCmd(
+			cmdnode.NodeFlags(),
+			p2p.Flags(),
+			core.Flags(),
+			cmdnode.MiscFlags(),
+			rpc.Flags(),
+			gateway.Flags(),
+			state.Flags(),
+		),
```
Comment (Wondertan) [inline cmd/celestia/bridge.go:46]: * Why authCmd needs all these flags? It seems like it doesn't
* AuthCmd should be located in `cmd` not `cmd/celestia`

**Review by Wondertan — CHANGES_REQUESTED** [review]
There is a bunch of places where we say permissions in plural while in fact, there is only once permission string. See cmd description, variables, and error messages.

File: cmd/celestia/bridge.go (hunk 1 above)
Comment (renaynay) [reply cmd/celestia/bridge.go:46]: It needs to be able to parse node store flags, p2p flags, etc. Also persistent pre run on bridge, light and full cmds.

**Review by distractedm1nd — APPROVED** [review]
excellent

**Review by Wondertan — APPROVED** [review]

**Review by liamsi — APPROVED** [review]
Only reviewed the tiny change in the ADR which LGTM

**Pull request merged**